### DISCUSSION TITLE
Pair Devices

### DISCUSSION THREAD

**Discussion by fruscil-jr** [dotnet-bluetooth-le/dotnet-bluetooth-le#797] [Q&A]
Hello,
I have a few different BLE devices that I want to programmatically pair using this library.
Calling the Adapter.BondAsync(IDevice) seems to hang on some devices, others it prompted to pair and worked fine (which is what I want :-)).

Q1.  Is this the correct method to call to pair devices?  I've also tried using ConnectToKnownDeviceAsync but for devices that aren't already paired the call just hangs. Any advance would be greatly appreciated.  Thanks in advance.

**Comment by axa88** [comment]
> Is this the correct method to call to pair devices?

It is the only method to Bond, and it only currently works on `Android`

> I've also tried using `ConnectToKnownDeviceAsync`

That is not for bonding, it is for connecting to a device.

>Calling the `Adapter.BondAsync(IDevice)` seems to hang on some devices

~~By "hang" what do you mean? 
Some devices are not connectable.  Depending on how the call is terminated it might throw, for example if it is canceled in some fashion.
Much more info is needed in order to help.~~

Yes bonding, still, isnt fully supported. The issue you were having is that the library was attempting pair and this library was ignoring the pair response... 
The latest version might work for you if you dont care how it gets paired. and the device your pairing with accepts minimal confirmation.

If you want full windows pairing support you can pick at this #906
